src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1029,6 +1029,23 @@ private void UpdateDatabase(TransactionOperationContext context, string type, Bl
                         }
 
                         databaseRecord = JsonDeserializationCluster.DatabaseRecord(databaseRecordJson);
+                        
+                        if (type == "AddRavenEtlCommand" || type == "AddSqlEtlCommand")
```

Comment:
This should be handled when you are putting new ETL. In `OutgoingTasksHandler.AddEtl`

Reply:
Also I think that when we want to remove connection string in `OutgoingTasksHandler.RemoveConnectionString` then we need to check if any ETLs are using it or not and throw exception with list of them when we find any.

Reply:
done